pandas/core/nanops.py

```diff
@@ -384,8 +384,12 @@ def _na_for_min_count(
     else:
         assert axis is not None  # assertion to make mypy happy
         result_shape = values.shape[:axis] + values.shape[axis + 1 :]
-        result = np.empty(result_shape, dtype=values.dtype)
-        result.fill(fill_value)
+        # calling np.full with dtype parameter throws an ValueError when called
+        # with np.datetime64 and pd.NaT
+        try:
+            result = np.full(result_shape, fill_value, dtype=values.dtype)
```

Comment:
what hits each of these branches?

Reply:
when `np.full` is called with the parameter `dtype=np.datetime64` and `fill_value=pd.NaT` a ValueError `ValueError: cannot convert float NaN to integer`. In this case I don't call `np.full` without an explicit `dtype`. Giving the array `result` an explicit dtype does not have an effect on the DataFrame column but I thought it is cleaner if I at least try to give it the correct one.